Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,291 @@
-﻿namespace BootCamp.Chapter
+﻿using System.Text;
+
+namespace BootCamp.Chapter
 {
     public static class BalanceStats
     {
+        private const string invalidMessage = "N/A.";
+        private const string currencySymbol = "¤";
+
         /// <summary>
         /// Return name and balance(current) of person who had the biggest historic balance.
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return invalidMessage;
+            }
+
+            float highestBalanceEver = HighestBalanceEver(peopleAndBalances);
+            string highestBalanceNames = BalanceNames(highestBalanceEver, peopleAndBalances);
+
+            return $"{highestBalanceNames} had the most money ever. {currencySymbol}{highestBalanceEver}.";
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return invalidMessage;
+            }
+            string[][] workingPeopleAndBalances = ArrayifyBalances(peopleAndBalances);
+            for (int i = 0; i < workingPeopleAndBalances.Length; i++)
+            {
+                if (workingPeopleAndBalances[i].Length <= 2)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5156603)